### ISSUE TITLE
Content sometimes disappears when moving back to Terminal window

### ISSUE THREAD

**Issue by wazzamatazz** [microsoft/terminal#2733]
# Environment

```none
Windows build number: 10.0.18362.0
Windows Terminal version (if applicable): 0.4.2382.0

Any other software?
```

# Steps to reproduce

Can't reproduce consistently, but sometimes, when I navigate back to Windows Terminal from another window, the content of the Terminal window completely disappears and the window becomes transparent.

![image](https://user-images.githubusercontent.com/7179614/64783745-3c3a8680-d571-11e9-852b-fc384117c519.png)

I'm using PowerShell Core, and I typically find that the problem occurs after running an `npm` script. Sometimes, if I hit up and then Enter, the previous command re-runs and the window returns to normal. On other occasions, this seems to have no effect.

**Comment by wazzamatazz** [comment]
Just happened again. The minimize/maximize/close buttons are unavailable when it happens - the disappear from the window. However, the new tab button was still active, and everything came back to life when I opened a new tab.

**Comment by encladeus** [comment]
As @wazzamatazz mentions above about the min/max/close buttons being unavailable (not displayed) when this issue exhibits, that jogged my memory and that is another symptom that I too have experienced.

**Comment by filip-hejsek** [comment]
I found a reliable way to reproduce this issue - minimize the terminal when a new tab is openning. You can do this by clicking on the `+` icon and immediately pressing <kbd>Win</kbd>+<kbd>D</kbd>

**Comment by dsafa** [comment]
Ran into this too. From initial testing using the above reproduction steps, it only occurs with the non client window so that's a good place to start looking.

**Comment by dsafa** [comment]
I figured out the issue. The non client window applies a window region to the xaml island to expose the caption area. The region is updated when the window is resized or the tab area size changes. If this is happening while the window is minimized, the small window size causes the region to be calculated as a rectangle that only exposes part of the tab area, hiding everything else.

![WindowsTerminal_qEaqScGrhp](https://user-images.githubusercontent.com/7334282/65263330-40125f80-dac1-11e9-8346-0ce963e4c78b.png)

Here you can see the shape of the window region. The red area is painted in the host window.

![WindowsTerminal_xpm10oZSel](https://user-images.githubusercontent.com/7334282/65263542-b31bd600-dac1-11e9-9705-15826f6d0799.png)

Here is the window region when it messes up. It only shows a small area of the tab bar. Not sure if these images illustrate the issue properly.

The only thing I'm not sure of is how this issue occurs when just switching between windows since that shouldn't cause a size change and the only cause I know for now is from minimizing. Maybe this is a different issue, but it looks the same to me.

**Comment by filip-hejsek** [comment]
You are right - the issue occurs if the size of the tab area changes while the window is minimized. This also happens when some program changes the tab title.

```bash
sleep 3; echo -en "\033]0;x\a"; read -sN 1
```
Running this script in wsl, minimizing the window, waiting a few seconds and restoring the window causes the content to disappear because the title is changed to "x". Pressing any key then changes the title back which causes the content to reappear.

**Comment by wazzamatazz** [comment]
> 
> 
> You are right - the issue occurs if the size of the tab area changes while the window is minimized. This also happens when some program changes the tab title.
> 
> ```shell
> sleep 3; echo -en "\033]0;x\a"; read -sN 1
> ```
> 
> Running this script in wsl, minimizing the window, waiting a few seconds and restoring the window causes the content to disappear because the title is changed to "x". Pressing any key then changes the title back which causes the content to reappear.

This probably makes sense with what I was seeing as well - I was running into the problem when running `npm` commands, which change the title at the start and end of the operation.

**Comment by mh-mazen** [comment]
![image](https://user-images.githubusercontent.com/35299529/67285159-bac70400-f4df-11e9-9726-868e9aa64639.png)

Issue Still Occurring on the latest preview as well
